packages/node/src/node.ts

```diff
@@ -1 +1,96 @@
-export default class Node {}
+import machine from "@counterfactual/machine";
+import * as ethers from "ethers";
+import EventEmitter from "eventemitter3";
+import firebase from "firebase";
+
+import {
+  Address,
+  AppInstanceInfo,
+  GetAppInstancesResult,
+  MethodName,
+  MethodRequest,
+  MethodResponse
+} from "./node-types";
+
+export default class Node {
+  // @ts-ignore
+  private instructionExecutor: machine.InstructionExecutor = Object.create(
+    null
+  );
+
+  /**
+   * Because the Node receives and sends out messages based on Event type
+   * https://github.com/counterfactual/monorepo/blob/master/packages/cf.js/src/types/node-protocol.ts#L21-L33
+   * the same EventEmitter can't be used since responses messages would get
+   * sent to listeners expecting a request message.
+   **/
+  private incoming: EventEmitter;
+  private outgoing: EventEmitter;
+
+  constructor(
+    // @ts-ignore
+    private readonly privateKey: string,
+    // @ts-ignore
+    private readonly provider: ethers.providers.Web3Provider,
```

Comment:
Should be the more generic type of provider; we will want to pass in `localhost:8545` for example